src/wg/format/bmp.d

```diff
@@ -9,10 +9,51 @@ License:    $(WEB boost.org/LICENSE_1_0.txt, Boost License 1.0)
 
 module wg.format.bmp;
 
-import wg.image.imagebuffer;
+import wg.image;
 import wg.image.metadata;
+import wg.image.transform;
 import wg.util.allocator;
 
+/**
+Create an ImageBuffer from a BMP formatted image.
+*/
+ImageBuffer readBMP(void[] bmpBuffer)
+{
+    auto bmp = BMPImage(bmpBuffer);
+    ImageBuffer src = bmp.getImage();
+    return src.clone();
```

Comment:
Why we need to clone here and not just return src?

Reply:
src is not an image buffer, it's just pointers into a BMP file.
some encodings of BMP are a valid image buffer (but someone else owns the allocation), and others need to be decoded (compression).

i'll add a function like that later when `ImageSource` is defined what that means, then we could make a pattern for that sort of function.

Reply:
i mean, i figured the idea for `readBMP` is to return an image buffer that you own.